### PR TITLE
deprecate ncplane_new(), update all callers

### LINKED ISSUE #1115: deprecate ncplane_new()
Since the 1.7 series we've had this note in `notcurses.h`:

```
// This function will be marked deprecated in 2.0 in favor of ncplane_create().
// It persists only for backwards compatibility.
API struct ncplane* ncplane_new(struct ncplane* n, int rows, int cols, int y, int x, void* opaque, const char* name);
```

Let's go ahead and mark it as deprecated, then, and remove it for 3.0.

### PR DESCRIPTION
**Pull request by dankamongmen**
Deprecate `ncplane_new()` in favor of `ncplane_create()`. Update all callers (core, wrappers, PoCs, tests, demo, view, fetch). Update `notcurses_ncplane.3` man page. Update `USAGE.md`.

Eliminate the `horiz` union of `ncplane_options`. It wasn't providing any actual typing benefit.

Closes #1115.

### CODE REVIEW

**Comment by dankamongmen** [comment]
We clearly need to touch

```
/// The `horiz` union field of [`NcPlaneOptions`](type.NcPlaneOptions)
///
/// It contains the fields:
///
/// - x: i32
/// - align: NcAlign (u32)
///
pub type NcPlaneOptionHoriz = crate::ncplane_options__bindgen_ty_1;
```

**Comment by dankamongmen** [comment]
OK I've taken a swing at adapting the rust, @joseluis

File: rust/src/plane.rs [hunk 1]
```diff
@@ -176,30 +176,25 @@ use crate::{
 impl NcPlaneOptions {
     /// `NcPlaneOptions` simple constructor with horizontal x
     pub fn new(y: i32, x: i32, rows: u32, cols: u32) -> Self {
-        Self::with_all_options(y, NcHoriz::x(x), rows, cols, 0)
+        Self::with_all_options(y, x, rows, cols, 0)
     }
 
     /// `NcPlaneOptions` simple constructor with horizontal alignment
-    pub fn new_halign(y: i32, align: NcAlign, rows: u32, cols: u32) -> Self {
+    pub fn new_halign(y: i32, x: i32, rows: u32, cols: u32) -> Self {
```
Comment (joseluis) [inline rust/src/plane.rs:183]: This should be `align: NcAlign`,

File: rust/src/plane.rs [hunk 2]
```diff
@@ -176,30 +176,25 @@ use crate::{
...
         Self::with_all_options(
             y,
-            NcHoriz::align(align),
+            x,
```
Comment (joseluis) [inline rust/src/plane.rs:186]: `align`,

File: rust/src/types/plane.rs [hunk 3]
```diff
@@ -12,24 +12,9 @@ pub type NcPlane = crate::ncplane;
 /// Options struct for [`NcPlane`](type.NcPlane.html)
 pub type NcPlaneOptions = crate::ncplane_options;
 
-/// Horizontal alignment relative to the parent plane. Use 'align' instead of 'x'.
+/// Horizontal alignment relative to the parent plane. Set alignment in 'x'.
```
Comment (joseluis) [inline rust/src/types/plane.rs:15]: I'll probably reword is in the future

File: rust/src/plane.rs (hunk 1 above)
Comment (dankamongmen) [reply rust/src/plane.rs:183]: Done, thanks.

File: rust/src/plane.rs (hunk 2 above)
Comment (dankamongmen) [reply rust/src/plane.rs:186]: Done, thanks.

**Review by joseluis — CHANGES_REQUESTED** [review]
Everything ok.  Just the `new_halign()` constructor needs to use the parameter ´align: NcAlign´ instead of `x: i32`

File: rust/src/plane.rs [hunk 4]
```diff
@@ -180,21 +180,21 @@ impl NcPlaneOptions {
...
+            align,
             rows,
             cols,
             NCPLANE_OPTION_HORALIGNED,
         )
     }
 
     /// `NcplaneOptions` constructor
-    pub fn with_all_options(y: i32, x: i32, rows: u32, cols: u32, flags: u64) -> Self {
+    pub fn with_all_options(y: i32, align: NcAlign, rows: u32, cols: u32, flags: u64) -> Self {
```
Comment (joseluis) [inline rust/src/plane.rs:194]: not in this function, revert to `x: i32`

File: rust/src/plane.rs [hunk 5]
```diff
@@ -180,21 +180,21 @@ impl NcPlaneOptions {
     }
 
     /// `NcPlaneOptions` simple constructor with horizontal alignment
-    pub fn new_halign(y: i32, x: i32, rows: u32, cols: u32) -> Self {
+    pub fn new_halign(y: i32, align: NcAlign, rows: u32, cols: u32) -> Self {
         Self::with_all_options(
             y,
-            x,
+            align,
```
Comment (joseluis) [inline rust/src/plane.rs:186]: sorry,

here `align: align as i32,` I think...

**Comment by dankamongmen** [comment]
OK, it looks like we have a clean build once more. Gonna go ahead and merge this.

File: rust/src/plane.rs (hunk 4 above)
Comment (dankamongmen) [reply rust/src/plane.rs:194]: Done, but did you see that it wanted `x` as `u32` in `new()` above? Is that expected?

File: rust/src/plane.rs (hunk 5 above)
Comment (dankamongmen) [reply rust/src/plane.rs:186]: Done, thanks.

File: rust/src/plane.rs (hunk 4 above)
Comment (dankamongmen) [reply rust/src/plane.rs:194]: Nevermind, got it, my dumbness.

**Comment by dankamongmen** [comment]
> Everything ok. Just the `new_halign()` constructor needs to use the parameter ´align: NcAlign´ instead of `x: i32`

OK, I think I got it right now. PTAL prior to merge?

**Review by joseluis — APPROVED** [review]
LGTM

**Pull request merged**